gammapy/catalog/hess.py

```diff
@@ -693,6 +741,41 @@ def gaussian_component(self, row_idx):
         data[SourceCatalogObject._row_index_key] = row_idx
         return SourceCatalogObjectHGPSComponent(data=data)
 
+    def to_models(self, which="best", components_status="independent"):
+        """ Create Models object from catalogue
+
+        Parameters
+        ----------
+        which : {'best', 'pl', 'ecpl'}
+            Which spectral model
+            
+        components_status : {'independent', 'linked', 'merged'}
+            Relation between the sources components:
+                'independent' : each sub-component of a source is given as 
+                                a diffrent `SkyModel` (Default)
+                'linked' : each sub-component of a source is given as 
+                           a diffrent `SkyModel` but the spectral parameters
+                           except the mormalisation are linked.
+                'merged' : the sub-components are merged into a single `SkyModel`
+                           given as a `~gammapy.modeling.models.TemplateSpatialModel`
+                           with a `~gammapy.modeling.models.PowerLawNormSpectralModel`.
+                           In that case the relave weigths between the components
+                           cannot be adjusted.
+
+        Returns
+        -------
+        models : `~gammapy.modeling.models.Models`
+            Models of the catalog.
+        """
+
+        models = []
+        for _ in self:
+            m = _.sky_model(which, components_status)
```

Comment:
It's always the better pattern to pass arguments by name and not rely on the order, because the order might change when a function is refactored in future. So I recommend to use `.sky_model(which=which, components_status=component_status)`